### PR TITLE
Fire pre harm and post harm events from [harm_unit] (Issue #8974)

### PR DESCRIPTION
**Pull request by rodolphecombe**
https://github.com/wesnoth/wesnoth/issues/8974
[harm_unit] now fires a pre harm and a post harm event, checking after each of those that the harmed unit still exists.

### CODE REVIEW

**Comment by soliton-** [comment]
Instead of overloading `fire_event` even more it'd be better to add a `fire_harm_events` attribute IMO.

**Comment by ProditorMagnus** [comment]
If we use harm_unit-specific event names, then events can always fire without checking fire_event.

File: data/lua/wml/harm_unit.lua [hunk 1]
```diff
@@ -37,198 +37,220 @@ function wml_actions.harm_unit(cfg)
...
-					}
+			
+			if fire_event then
+				local harmer_loc={0, 0}
+				local unit_to_harm_loc={ x = tonumber(unit_to_harm.x) or 0, y = tonumber(unit_to_harm.y) or 0 }
+				if harmer and harmer.valid then
+					harmer_loc = { x = tonumber(harmer.x) or 0, y = tonumber(harmer.y) or 0 }
 				end
-				wesnoth.interface.scroll_to_hex(unit_to_harm.x, unit_to_harm.y, true)
+				wesnoth.game_events.fire("pre harm", unit_to_harm_loc, harmer_loc, primary_attack, secondary_attack)
```
Comment (CelticMinstrel) [inline data/lua/wml/harm_unit.lua:47]: This is incorrect, the attacks need to be enclosed in a WML table: `{primary_attack = primary_attack, secondary_attack = secondary_attack}`

Comment (rodolphecombe) [reply data/lua/wml/harm_unit.lua:47]: Like that?
`wesnoth.game_events.fire("pre harm", unit_to_harm_loc, harmer_loc, {primary_attack = primary_attack, secondary_attack = secondary_attack, amount=amount})`
What would damage_inflicted value? amount for pre harm and damage for post harm?
https://github.com/wesnoth/wesnoth/blob/944ed026a255e996a571ee5bbc357f7d1b5d961f/data/lua/wml/harm_unit.lua#L88C4-L94C5

**Comment by Pentarctagon** [comment]
Closing given the review comments and lack of response. @rodolphecombe Feel free to re-open this PR if you decide to come back to it.

**Comment by rodolphecombe** [comment]
No problem. I am waiting for some feedback from EmperorZombie they was supposed to test if my implementation fits their needs. 
The commit is ready but I have not yet updated the PR

**Comment by rodolphecombe** [comment]
@Pentarctagon I looks like I either cannot reopen the PR myself and do not know how to do it. Could you please do it for me? I would like to submit the validated version of the code.

File: data/lua/wml/harm_unit.lua [hunk 2]
```diff
@@ -37,198 +37,230 @@ function wml_actions.harm_unit(cfg)
...
-			if kill ~= false and unit_to_harm.hitpoints <= 0 then
-				wml_actions.kill { id = unit_to_harm.id, animate = toboolean( animate ), fire_event = fire_event, harmer and T.secondary_unit { id = harmer.id } }
-			end
+				local function calc_xp( level ) -- to calculate the experience in case of kill
+					if level == 0 then return math.ceil(wesnoth.game_config.kill_experience / 2)
+					else return level * wesnoth.game_config.kill_experience end
+				end
+									
+				if fire_event then
+					local harmer_loc={0, 0}
```
Comment (CelticMinstrel) [inline data/lua/wml/harm_unit.lua:212]: For consistency:
```suggestion
					local harmer_loc={ x = 0, y = 0}
```

**Comment by CelticMinstrel** [comment]
My overall stance on this hasn't change – I don't see a point in adding a new event solely for a WML tag.

**Comment by rodolphecombe** [comment]
So be it then.
At least the PR with the working code will be here if someone see more interest in adding that later on. 
I do not think it will be hard to merge later.

**Comment by Pentarctagon** [comment]
Re-closing in the mean time then.

**Pull request closed without merging**